src/bin/cli.rs

```diff
@@ -93,6 +107,26 @@ async fn main() -> mini_redis::Result<()> {
             client.set_expires(&key, value, expires).await?;
             println!("OK");
         }
+        Command::Publish { channel, message } => {
+            client.publish(&channel, message.into()).await?;
+            println!("Publish OK");
+        }
+        Command::Subscribe { channels } => {
+            if channels.is_empty() {
+                return Err("channel(s) must be provided".into());
+            }
+            let mut subscriber = client.subscribe(channels).await?;
+
+            loop {
+                // await messages on channels
+                if let Some(msg) = subscriber.next_message().await? {
```

Comment:
We don't want to busy-loop if the connection closes.
```suggestion
            while let Some(msg) = subscriber.next_message().await? {
```